src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/JQueryQueryStringValueProviderFactory.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Internal;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding
+{
+    /// <summary>
+    /// An <see cref="IValueProviderFactory"/> for <see cref="JQueryQueryStringValueProvider"/>.
+    /// </summary>
+    public class JQueryQueryStringValueProviderFactory : IValueProviderFactory
+    {
+        /// <inheritdoc />
+        public Task CreateValueProviderAsync(ValueProviderFactoryContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var request = context.ActionContext.HttpContext.Request;
+            var valueProvider = new JQueryQueryStringValueProvider(
```

Comment:
Don't create or add the value provider if the query string is empty.

Reply:
I was just following whatever we are doing in https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/QueryStringValueProviderFactory.cs
Let me know if you still want me to change it.

Reply:
Yes, please fix this and `QueryStringValueProviderFactory` to be as efficient as `FormValueProviderFactory` and `JQueryFormValueProviderFactory`. Both no-op when the request does not contain HTML form URL-encoded data.